src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -758,9 +758,20 @@ static int rbd_nbd(int argc, const char *argv[])
 
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
-			 CODE_ENVIRONMENT_DAEMON,
-			 CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
+
+  code_environment_t code_env = CODE_ENVIRONMENT_UTILITY;
+  int env_flag = 0;
+  for (auto arg : args) {
+    if (arg != NULL && arg[0] != '-') {
+      if (strcmp(arg, "map") == 0) {
+        code_env = CODE_ENVIRONMENT_DAEMON;
+        env_flag = CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS;
+      }
+      break;
+    }
+  }
+
+  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT, code_env, env_flag);
```

Comment:
@trociny, I tried rbd-nbd --admin-socket=/path/test.sock map rbd/test_image, and rbd-nbd list-mapped, and no error. I believe you tried the case which add admin-socket by config file. That is not hard to fix, I could add one line after global_init for list-mapped and unmap commands:
 `  g_ceph_context->_conf->set_val_or_die("pid_file", "");`
I believe this will resolve the issue.